redbeams/src/test/java/com/sequenceiq/redbeams/configuration/SslCertificateEntryTest.java

```diff
@@ -38,12 +43,37 @@ static Object[][] constructorTestWhenNPEDataProvider() {
     @ParameterizedTest(name = "{0}")
     @MethodSource("constructorTestWhenNPEDataProvider")
     void constructorTestWhenNPE(String testCaseName, int version, String cloudProviderIdentifier, String certPem, X509Certificate x509Cert) {
-        assertThrows(NullPointerException.class, () -> new SslCertificateEntry(version, cloudProviderIdentifier, certPem, x509Cert));
+        assertThrows(NullPointerException.class, () -> new SslCertificateEntry(version, cloudProviderIdentifier,
+                cloudProviderIdentifier, AWS, certPem, x509Cert));
+    }
+
+    static Object[][] constructorTestWhenNONNPEDataProvider() {
+        return new Object[][]{
+                // testCaseName version cloudProviderIdentifier certPem x509Cert
+                {"0, cloudProviderIdentifier, null, x509Cert", VERSION_0, CLOUD_PROVIDER_IDENTIFIER,
+                        CLOUD_PROVIDER_IDENTIFIER, CERT_PEM, X_509_CERT},
+                {"0, cloudProviderIdentifier, null, x509Cert", VERSION_0, CLOUD_PROVIDER_IDENTIFIER_DEFAULT,
+                        CLOUD_PROVIDER_IDENTIFIER, CERT_PEM, X_509_CERT},
+        };
+    }
```

Comment:
Perhaps add one more test case where `cloudProviderIdentifier` contains a region different than the `.default` suffix, e.g. `"certID.someregion"`. `getCloudKey()` shall then return the input `cloudProviderIdentifier` unchanged without stripping the suffix.